File: PHP/PetShop.php
<?php
class PetShop {
    public function __construct() {
        // Memulai session jika belum ada
        if (!isset($_SESSION['products'])) {
            $_SESSION['products'] = [];
        }
    }

    // Menambahkan produk baru
    public function addProduct($name, $category, $price, $image) {
        $product = [
            'id' => count($_SESSION['products']) + 1, // Menghasilkan ID secara otomatis
            'name' => $name,
            'category' => $category,
            'price' => $price,
            'image' => $image
        ];
        $_SESSION['products'][] = $product;
    }

    // Menampilkan produk
    public function displayProducts() {
        if (empty($_SESSION['products'])) {
            echo "<p>No products available.</p>";
            return;
        }

        echo "<table border='1'>
                <tr>
                    <th>ID</th>
                    <th>Name</th>
                    <th>Category</th>
                    <th>Price</th>
                    <th>Image</th>
                    <th>Actions</th>
                </tr>";

        foreach ($_SESSION['products'] as $product) {
            echo "<tr>
                    <td>{$product['id']}</td>
                    <td>{$product['name']}</td>
                    <td>{$product['category']}</td>
                    <td>\${$product['price']}</td>
                    <td><img src='images/{$product['image']}' width='50'></td>
                    <td>
                        <a href='?action=edit&id={$product['id']}'>Edit</a> |
                        <a href='?action=delete&id={$product['id']}'>Delete</a>
                    </td>
                  </tr>";
        }
        echo "</table>";
    }

    // Menghapus produk berdasarkan ID
    public function deleteProduct($id) {
        foreach ($_SESSION['products'] as $key => $product) {
            if ($product['id'] == $id) {
                unset($_SESSION['products'][$key]); // Hapus produk berdasarkan ID
                $_SESSION['products'] = array_values($_SESSION['products']); // Menata ulang array
                return;
            }
        }
    }

    // Mencari produk berdasarkan nama
    public function searchProduct($name) {
        $found = false;
        foreach ($_SESSION['products'] as $product) {
            if (stripos($product['name'], $name) !== false) {
                echo "<p>Found: {$product['name']} ({$product['category']}) - \${$product['price']}</p>";
                $found = true;
            }
        }
        if (!$found) {
            echo "<p>No product found with the name '$name'.</p>";
        }
    }

    // Mengubah produk berdasarkan ID
    public function updateProduct($id, $newName, $newCategory, $newPrice, $newImage) {
        foreach ($_SESSION['products'] as &$product) {
            if ($product['id'] == $id) {
                $product['name'] = $newName;
                $product['category'] = $newCategory;
                $product['price'] = $newPrice;
                $product['image'] = $newImage;
                return;
            }
        }
    }
}
?>

File: Java/PetShop.java
// PetShop.java
import java.util.ArrayList;
import java.util.Scanner;

class PetShop {
    private static class Product {
        private int id;
        private String name;
        private String category;
        private double price;

        public Product(int id, String name, String category, double price) {
            this.id = id;
            this.name = name;
            this.category = category;
            this.price = price;
        }

        public int getId() { return id; }
        public String getName() { return name; }
        public String getCategory() { return category; }
        public double getPrice() { return price; }

        public void setName(String name) { this.name = name; }
        public void setCategory(String category) { this.category = category; }
        public void setPrice(double price) { this.price = price; }

        public String toString() {
            return "ID: " + id + ", Name: " + name + ", Category: " + category + ", Price: " + price;
        }
    }

    private ArrayList<Product> products = new ArrayList<>();
    private int nextId = 1;

    public void displayProducts() {
        if (products.isEmpty()) {
            System.out.println("No products available.");
        } else {
            for (Product p : products) {
                System.out.println(p);
            }
        }
    }

    public void addProduct(String name, String category, double price) {
        products.add(new Product(nextId++, name, category, price));
    }

    public void updateProduct(int id, String name, String category, double price) {
        for (Product p : products) {
            if (p.getId() == id) {
                p.setName(name);
                p.setCategory(category);
                p.setPrice(price);
                return;
            }
        }
        System.out.println("Product not found.");
    }

    public void deleteProduct(int id) {
        products.removeIf(p -> p.getId() == id);
    }

    public void searchProduct(String name) {
        for (Product p : products) {
            if (p.getName().equalsIgnoreCase(name)) {
                System.out.println(p);
                return;
            }
        }
        System.out.println("Product not found.");
    }
}
